### DISCUSSION TITLE
Miryoku-Kmonad, `Ctrl + Shift + P` not working if using Right hand to press `Ctrl + Shift` and using left hand to press `P` - colemak layout.

### DISCUSSION THREAD

**Discussion by KAGEYAM4** [manna-harbour/miryoku#274] [Q&A] [answered]
Colemak mod dh layout.

As stated in title, if i use my right hand for `ctrl+shift` and left hand to press `p`, nothing happens.
But if i use my left hand for all three `ctrl+shift+p`, this works.

For referende, this is my keyboard ->

![output](https://github.com/manna-harbour/miryoku/assets/75798544/889f575e-36b5-42fa-ada0-a668ca546371)

Left Home row - `A,R,S,T`
Right Home row - `N,E,I,O`

Left Thumb Home row - `2,3,4`
Right Thumb Home row - `9, -, 0`

**Comment by manna-harbour** [comment, marked as answer]
Some regular keyboards don't have sufficient key rollover outside the normal modifiers. See https://en.wikipedia.org/wiki/Key_rollover.

You can try with a keyboard tester that shows press and release events. See  https://github.com/samhocevar-forks/qmk-firmware/blob/master/docs/newbs_testing_debugging.md#testing.

**Comment by KAGEYAM4** [comment]
> Some regular keyboards don't have sufficient key rollover outside the normal modifiers. See [en.wikipedia.org/wiki/Key_rollover](https://en.wikipedia.org/wiki/Key_rollover).
> 
> You can try with a keyboard tester that shows press and release events. See [samhocevar-forks/qmk-firmware@`master`/docs/newbs_testing_debugging.md#testing](https://github.com/samhocevar-forks/qmk-firmware/blob/master/docs/newbs_testing_debugging.md#testing).

I tested on https://config.qmk.fm/#/test/, I held `ctrl` and `shift` and pressed all other keys. I could see that all three keys are being registered.


But when it came to `Ctrl+shift` on right hand , `p` on left hand, `p` wasn't getting registered.

Could this be a bug in kmonad or config?

**Reply by manna-harbour** [reply]
Try the same physical keys without KMonad.

**Comment by KAGEYAM4** [comment]
> Try the same physical keys without KMonad.

i turned off kmonad, 

--- these three-keys are not getting registered together, only two out of three works.

Does this mean my keyboard has prolem, could this have been caused physically as you can see - i changed the keys up and down and in this process broke key-lock of two keys.

or could this be msi-firmware issue?

**Reply by manna-harbour** [reply]
It's by design. See https://en.wikipedia.org/wiki/Key_rollover#Multi-key_rollover.

**Reply by manna-harbour** [reply]
You might have better luck with a different home position, say one place further out. Otherwise you'd need to use a different keyboard.

**Reply by KAGEYAM4** [reply]
Ok, got it. Thank you for the help.
